modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -751,9 +751,10 @@ public BVarSymbol defineVarSymbol(DiagnosticPos pos, Set<Flag> flagSet, BType va
 
         // Add it to the enclosing scope
         // Find duplicates
-        if (symResolver.checkForUniqueSymbol(pos, env, varSymbol, SymTag.VARIABLE_NAME)) {
-            enclScope.define(varSymbol.name, varSymbol);
+        if (!symResolver.checkForUniqueSymbol(pos, env, varSymbol, SymTag.VARIABLE_NAME)) {
+            varSymbol.type = symTable.errType;
```

Comment:
Even if the variable is redeclared, it has a type and we should use that type for subsequent expression analysis. Why did we change that?

Reply:
@sameera-jayasoma. It causes to report uncertain errors for the following scenario. 

```
string error = "abc";

function main(string[] args){
    error err = { txt : "abc"};
}

struct error {
    string txt;
}
```
Following is the compilation result. 
```
error: pkg.bal:7:1: redeclared builtin symbol 'error'
error: pkg.bal:1:1: redeclared builtin symbol 'error'
error: pkg.bal:4:19: undefined field 'txt' in struct 'error'
error: pkg.bal:1:1: undefined symbol 'error'
```
Here error 3 is reported by resolving the builtin error, But that error depends on the program, so sometimes it can be correct, but sometimes not.  Additional it reports 4th error `undefined symbol 'error'` which is not correct. 

If the symbol is not unique, I have marked that symbol's type as an error type and then define. This will avoid uncertain errors. 

with this change following is the new compilation result. 

```
error: pkg.bal:7:1: redeclared builtin symbol 'error'
error: pkg.bal:1:1: redeclared builtin symbol 'error'
```